src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -73,4 +77,44 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
                 return false;
         }
     }
+
+    public async isFirewallConfigured(): Promise<boolean> {
+        const resourceGroup: string = azureUtils.getResourceGroupFromId(this.id);
+        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+        const serverName: string = nonNullProp(this.server, 'name');
+        const ip: string = await publicIp.v4();
+
+        try {
+            const existingFirewallRule: FirewallRule = await client.firewallRules.get(resourceGroup, serverName, this.firewallRuleName);
+            return existingFirewallRule && existingFirewallRule.startIpAddress === ip && existingFirewallRule.endIpAddress === ip;
+        } catch (error) {
+            if (parseError(error).errorType === 'ResourceNotFound') {
+                // The firewall rule is not configured yet
+                return false;
+            } else {
+                throw error;
+            }
+        }
+    }
+
+    public async configureFirewall(): Promise<void> {
```

Comment:
Since you already have a `configurePostgresFirewall.ts` file - I would prefer to put all of this logic there. The tree files can grow pretty quickly if we're not careful 🌳🌳🌳🌳🌳🌳🌳🌳🌳